pkg/ext/stores/useractivity/store.go

```diff
@@ -0,0 +1,258 @@
+
+// GetSingularName implements [rest.SingularNameProvider]
+func (t *Store) GetSingularName() string {
+	return SingularName
+}
+
+// New implements [rest.Storage]
+func (t *Store) New() runtime.Object {
+	obj := &ext.UserActivity{}
+	obj.GetObjectKind().SetGroupVersionKind(GVK)
+	return obj
+}
+
+// Destroy implements [rest.Storage]
+func (t *Store) Destroy() {
+}
+
+// Create implements [rest.Creator]
+func (uas *Store) Create(ctx context.Context,
+	obj runtime.Object,
+	createValidation rest.ValidateObjectFunc,
+	options *metav1.CreateOptions) (runtime.Object, error) {
+
+	// retrieving useractivity object from raw data
+	objUserActivity, ok := obj.(*ext.UserActivity)
+	if !ok {
+		return nil, fmt.Errorf("error casting runtime object to useractivity")
+	}
+	// retrieve token information
+	token, err := uas.tokenController.Get(objUserActivity.Spec.TokenId, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get token %s: %v", objUserActivity.Spec.TokenId, err)
+	}
+	// set when last activity happened
+	lastActivity := metav1.Time{
+		Time: time.Now().UTC(),
+	}
+	// retrieve setting for auth-user-session-idle-ttl-minutes
+	idleTimeout := settings.AuthUserSessionIdleTTLMinutes.GetInt()
+
+	return uas.create(ctx, objUserActivity, token, token.UserID, lastActivity, idleTimeout)
+}
+
+// create sets the LastActivity and CurrentTimeout fields on the UserActivity object
+// provided by the user within the request.
+func (uas *Store) create(_ context.Context,
+	userActivity *ext.UserActivity,
+	token *v3Legacy.Token,
+	user string,
+	lastActivity metav1.Time,
+	authUserSessionIdleTTLMinutes int) (*ext.UserActivity, error) {
+
+	expectedName, err := setUserActivityName(user, token.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to set useractivity name: %v", err)
+	}
+	// ensure the UserActivity object is crafted as expected.
+	if userActivity.Name != expectedName {
+		return nil, fmt.Errorf("useractivity name provided mismatches from expected: %s - %s", userActivity.Name, expectedName)
+	}
+	// ensure the token specified in the UserActivity is the same
+	// we are using to do the request.
+	if token.Name != userActivity.Spec.TokenId {
+		return nil, fmt.Errorf("token name provided mismatches from expected: %s - %s", token.Name, userActivity.Spec.TokenId)
+	}
+
+	// once validated the request, we can define the lastActivity time.
+	newIdleTimeout := metav1.Time{
+		Time: lastActivity.Add(time.Minute * time.Duration(authUserSessionIdleTTLMinutes)).UTC(),
+	}
+	token.LastIdleTimeout = newIdleTimeout
+	userActivity.Status.LastActivity = lastActivity.String()
+	userActivity.Status.CurrentTimeout = newIdleTimeout.String()
+	_, err = uas.tokenController.Update(token)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update token: %v", err)
+	}
+
+	return userActivity, nil
+}
+
+// The rest of the methods will be left empty.
+
+// Delete implements [rest.GracefulDeleter]
+func (uas *Store) Delete(ctx context.Context,
```

Comment:
Remove this method since it is not supported. If you don't implement the `rest.GracefulDeleter` interface `delete` will not be available.
you can check available operations for ext resources with curl

```
curl -H "Authorization: Bearer token" https://host/ext/apis/ext.cattle.io/v1
```